src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -2262,6 +2263,34 @@ static u_int tunnel_counter_key_hash(const void *key) {
 						  chunk_hash(chunk_from_str((char*)tunnel_key->vrf_name)))));
 }
 
+static struct tunnel_counter_key *create_new_tunnel_counter_key(uint32_t terminator_id,
+																const char* tenant_id,
+																const char* account_id,
+																const char* vrf_name){
+	size_t tenant_id_len = strlen(tenant_id);
+	size_t account_id_len = strlen(account_id);
+	size_t vrf_name_len = strlen(vrf_name);
+	struct tunnel_counter_key *key = (struct tunnel_counter_key *)malloc(sizeof(struct tunnel_counter_key) +
+																		tenant_id_len + 1 +
+																		account_id_len + 1 +
+																		vrf_name_len + 1);
+	key->terminator_id = terminator_id;
+	key->tenant_id = (char*)(key + sizeof(struct tunnel_counter_key));
+	key->account_id = key->tenant_id + tenant_id_len + 1;
+	key->vrf_name = key->account_id + account_id_len + 1;
+	strcpy(key->tenant_id, tenant_id);
+	strcpy(key->account_id, account_id);
+	strcpy(key->vrf_name, vrf_name);
+	key->count = 0;
+
+	return key;
+}
+
+static void destroy_tunnel_counter_key(void *value, const void *key) {
+	(void)value;
+	free(key);
```

Comment:
Better for `free(value)` instead since `key` is a pointer to const and `free()` expects a pointer to non-const. This only compiles because the code base is lax in regards to const conversions, but no reason to continue with this is no need to and it's simple to fix it here